Updated to v4 API

geocodefarm opened the pull request:
Updated everything except for the tests to v4 API. Tests will need to be updated afterwords. Most should work, but some assume no API key is required and will need to be removed or updated.

geocodefarm committed:
Update README.md

Updated README to the new API specs for v4 and planned changes to rest of codebase

geocodefarm committed:
Update Changes

Added Geocode.Farm's changes to changelog

geocodefarm committed:
Update Changes

Added Geocode.Farm's changes to changelog + raised version number

geocodefarm committed:
Update README

Updated README

geocodefarm committed:
Update geocode.pl

Fixed to Require API Key

geocodefarm committed:
Update reverse_geocode.pl

Updated to require API key

geocodefarm committed:
Update GeocodeFarm.pm

Updated to use Geocode.Farm v4 API

geocodefarm committed:
Update GeocodeFarm.pm

Fixed reverse processing for v4 API

geocodefarm commented:
@dex4er

dex4er commented:
@geocodefarm Do you want to adapt this package? You can also wait a while if I'll make a proper Github Actions for this package. I need a day or two.

geocodefarm commented:
@dex4er We've updated what we could find it in. But we'd really need your help to get it fully adapted and updated. If you need anything to get it done, such as API key for testing and so on, let us know. Also none of our team are perl programmers, so we have no way to really test it and make sure it's all updated properly and working.

dex4er commented:
@geocodefarm Ok, so thanks for you changes. I'll review them and combine with all other changes as this library is not yet converted to Github Actions as all my other Perl repos.
